pkg/operator/controller/dns_status_test.go

```diff
@@ -0,0 +1,379 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDNSStatusConditions(t *testing.T) {
+	testCases := []struct {
+		description                                string
+		clusterIP                                  string
+		avail, desire                              int32
+		degradeType, progressType, availType       string
+		degradeStatus, progressStatus, availStatus operatorv1.ConditionStatus
+	}{
+		{
+			description:    "no cluster ip, 0/0 pods available",
+			clusterIP:      "",
+			avail:          0,
+			desire:         0,
+			degradeType:    operatorv1.OperatorStatusTypeDegraded,
+			progressType:   operatorv1.OperatorStatusTypeProgressing,
+			availType:      operatorv1.OperatorStatusTypeAvailable,
+			degradeStatus:  operatorv1.ConditionTrue,
+			progressStatus: operatorv1.ConditionFalse,
+			availStatus:    operatorv1.ConditionFalse,
+		},
+		{
+			description:    "no cluster ip, 0/2 pods available",
+			clusterIP:      "",
+			avail:          0,
+			desire:         2,
+			degradeType:    operatorv1.OperatorStatusTypeDegraded,
+			progressType:   operatorv1.OperatorStatusTypeProgressing,
+			availType:      operatorv1.OperatorStatusTypeAvailable,
+			degradeStatus:  operatorv1.ConditionTrue,
+			progressStatus: operatorv1.ConditionTrue,
+			availStatus:    operatorv1.ConditionFalse,
+		},
+		{
+			description:    "no cluster ip, 1/2 pods available",
+			clusterIP:      "",
+			avail:          1,
+			desire:         2,
+			degradeType:    operatorv1.OperatorStatusTypeDegraded,
+			progressType:   operatorv1.OperatorStatusTypeProgressing,
+			availType:      operatorv1.OperatorStatusTypeAvailable,
+			degradeStatus:  operatorv1.ConditionTrue,
+			progressStatus: operatorv1.ConditionTrue,
+			availStatus:    operatorv1.ConditionTrue,
```

Comment:
This looks odd—is DNS really available if it has no IP address?

Reply:
Good point. The previous "available" condition logic only considered having 1 running pod in the daemonset for the condition to be true. Let me update that logic to consider the clusterIP too.

Reply:
I will update the available condition to consider the ClusterIP status.